lib/index.js

```diff
@@ -96,16 +96,38 @@ function Socket (type, opts) {
             break;
     }
 
-    this.binding = nn.Socket(this.af_domain, this.protocol);
-    this.queue = [];
+  this.binding = nn.Socket(this.af_domain, this.protocol);
+  this.queue = [];
 
-    for(var sokopt in sol){
-        if(opts.hasOwnProperty(sokopt)) this[sokopt](opts[sokopt]);
+  /* subscription filters */
+  this.channels = {};
+
+  /* subscription handling at initialization */
+  if (opts.hasOwnProperty('chan')) {
+    if (typeof opts.chan == 'string') {
+      opts.chan.split(',').forEach(subscriber.bind(this))
+    } else {
+      opts.chan.forEach(subscriber.bind(this))
     }
+  }
 
-    if(this.af_domain == nn.AF_SP) {
-        if (this.receiver) this._startPollReceive();
+  function subscriber(chan) {
+    if (nn.Chan(this.binding, nn.NN_SUB_SUBSCRIBE, chan) != -1) {
+      this.channels[chan] = true;
+    } else {
+      console.log('sub err')
```

Comment:
I was thinking maybe we should throw here actually.

One thing that will come up in another PR (forthcoming related to `nn_shutdown()`), is that we'll need to decide about dynamic re-subscription, according to [folks on the mailing list](http://www.freelists.org/post/nanomsg/Question-on-pubsub-device,5) it may not be possible, and backing that idea up, I'm seeing some unverified issues with latent calls after `socket.rmchan()` on a specific topic.